### PR TITLE
added extension method to add Vocabulary Terms

### PR DESCRIPTION
**Pull request by xtofs**
extension method to add Vocabulary Terms analog to AddComplexType and AddEntityType

### Issues

No issues, just adding more of the types extensions analog to others.

### Description

Added two extension method to add Vocabulary Terms that wrap the two main Microsoft.OData.Edm.Vocabularies.EdmTerm constructors

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 1]
```diff
@@ -1260,6 +1260,34 @@ public static EdmEntityContainer AddEntityContainer(this EdmModel model, string
...
+        /// <summary>
+        /// Creates and adds a new instance of the Microsoft.OData.Edm.Vocabularies.EdmTerm class with a primitive type.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="namespaceName">Namespace of the term.</param>
+        /// <param name="type">The primitive type of the term.</param>
+        /// <returns>The created term.</returns>
+        public static EdmTerm AddTerm(this EdmModel model, string namespaceName, EdmPrimitiveTypeKind type)
+        {
+            var term = new EdmTerm(namespaceName, name, type);
```
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >name [](start = 50, length = 4)

where comes the "name"?

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 2]
```diff
@@ -1260,6 +1260,34 @@ public static EdmEntityContainer AddEntityContainer(this EdmModel model, string
             return container;
         }
 
+        /// <summary>
+        /// Creates and adds a new instance of the Microsoft.OData.Edm.Vocabularies.EdmTerm class with a primitive type.
```
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >Microsoft.OData.Edm.Vocabularies.EdmTerm [](start = 51, length = 40)

use <see cref="EdmTerm"/>

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 3]
```diff
@@ -1260,6 +1260,34 @@ public static EdmEntityContainer AddEntityContainer(this EdmModel model, string
...
+        /// <summary>
+        /// Creates and adds a new instance of the Microsoft.OData.Edm.Vocabularies.EdmTerm class.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="namespaceName">Namespace of the term.</param>
+        /// <param name="name">The type of the term.</param>
+        /// <returns>The created term.</returns>
+        public static EdmTerm AddTerm(this EdmModel model, string namespaceName, IEdmTypeReference type)
+        {
+            var term = new EdmTerm(namespaceName, name, type);
```
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >name [](start = 50, length = 4)

where comes the "name"

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 4]
```diff
@@ -1260,6 +1260,34 @@ public static EdmEntityContainer AddEntityContainer(this EdmModel model, string
...
+            model.AddElement(term);
+            return term;
+        }
+
+        /// <summary>
+        /// Creates and adds a new instance of the Microsoft.OData.Edm.Vocabularies.EdmTerm class.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="namespaceName">Namespace of the term.</param>
+        /// <param name="name">The type of the term.</param>
```
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >name">The type of the term [](start = 25, length = 26)

where comes the "name"?

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 5]
```diff
@@ -1260,6 +1260,34 @@ public static EdmEntityContainer AddEntityContainer(this EdmModel model, string
...
+        /// <returns>The created term.</returns>
+        public static EdmTerm AddTerm(this EdmModel model, string namespaceName, EdmPrimitiveTypeKind type)
+        {
+            var term = new EdmTerm(namespaceName, name, type);
+            model.AddElement(term);
+            return term;
+        }
+
+        /// <summary>
+        /// Creates and adds a new instance of the Microsoft.OData.Edm.Vocabularies.EdmTerm class.
```
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: >Microsoft.OData.Edm.Vocabularies.EdmTerm [](start = 51, length = 40)

use <see cref="EdmTerm"/>

**Comment by xuzhg** [comment]
@Christof Sprenger Thanks for your contribution. I left some comments, would you please take a look? Besides, would you please add some test cases to cover your changes?

**Comment by chrisspre** [comment]
@xuzhg happy to add test cases. Can you please point me to test cases for other of these sort of extension methods that wrap calls to `model.AddElement` in a more strongly typed way?

**Comment by xuzhg** [comment]
@chrisspre https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Microsoft.OData.Edm.Tests/ExtensionMethods/ExtensionMethodTests.cs maybe a right place to add.

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 2 above)
Comment (xtofs) [reply src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: fixed (copy/paste error :-(  )

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 3 above)
Comment (xtofs) [reply src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: fixed (copy/paste error :-(  )

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 5 above)
Comment (xtofs) [reply src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: fixed in last commit (# 3)

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 4 above)
Comment (xtofs) [reply src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: fixed (copy/paste error :-(  )

**Comment by xtofs** [comment]
The failing tests don't seem to be related to the change.

any advice how to proceed?

Failing tests: 

Failed	PublicApiTest	Microsoft.Test.Taupo.OData.Common.Tests.PublicApi.ODataPublicApiTests.PublicApiTest	Test method Microsoft.Test.Taupo.OData.Common.Tests.PublicApi.ODataPublicApiTests.PublicApiTest threw exception: ...	
Failed	EFActionTest	AstoriaUnitTests.Tests.Server.ActionTests.EFActionTest	Test method AstoriaUnitTests.Tests.Server.ActionTests.EFActionTest threw exception: ...	
Failed	EFActionTestWithPaging	AstoriaUnitTests.Tests.Server.ActionTests.EFActionTestWithPaging	Test method AstoriaUnitTests.Tests.Server.ActionTests.EFActionTestWithPaging threw exception: ...	
Failed	EFProviderNegativeTests	AstoriaUnitTests.Tests.Server.ActionTests.EFProviderNegativeTests	Test method AstoriaUnitTests.Tests.Server.ActionTests.EFProviderNegativeTests threw exception: ...	
Failed	EFProviderNegativeTests2	AstoriaUnitTests.Tests.Server.ActionTests.EFProviderNegativeTests2	Test method AstoriaUnitTests.Tests.Server.ActionTests.EFProviderNegativeTests2 threw exception: ...

**Comment by xuzhg** [comment]
PublicApiTest is related to your change, because you added two public apis.

Others seem not related to your changes.

**Comment by xtofs** [comment]
@xuzhg 
> PublicApiTest is related to your change, because you added two public apis.
> 
> Others seem not related to your changes.

the error comes from the assertion `Assert.IsTrue(Directory.Exists(testDirectory), "Test directory does not exists");` 

Theses test don't run in Visual Studio. how do I run them locally?
But nonetheless, i don't see why this fails on the test/build server based on my change. 

Any help how to move the pull request forward would be appreciated.

**Comment by xuzhg** [comment]
@xtofs Can you open the e2e solutions to run the test cases in your local VS?

**Comment by xuzhg** [comment]
merged at: 4b26ced34806884231476d3c4fb6f08926934eba

**Pull request closed without merging**